File: frontend/src/lib/categories.ts
// Phase 4 D-24/D-25: Hardcoded 10-category taxonomy mirroring
// src/NossoVizinho.Api/Constants/Categories.cs. Kept in sync manually.
// Cache from GET /api/v1/categories after first fetch to pick up admin toggles.

export interface LocalSubcategory {
  code: string;
  displayName: string;
}

export interface LocalCategory {
  code: string;
  displayName: string;
  subcategories: LocalSubcategory[];
}

export const CATEGORIES: LocalCategory[] = [
  {
    code: "eletronicos",
    displayName: "Eletrônicos & Informática",
    subcategories: [
      { code: "celular", displayName: "Celular" },
      { code: "notebook", displayName: "Notebook" },
      { code: "tv", displayName: "TV" },
      { code: "outros", displayName: "Outros" },
    ],
  },
  {
    code: "moveis",
    displayName: "Móveis & Decoração",
    subcategories: [
      { code: "sala", displayName: "Sala" },
      { code: "quarto", displayName: "Quarto" },
      { code: "cozinha", displayName: "Cozinha" },
    ],
  },
  {
    code: "roupas",
    displayName: "Roupas, Calçados & Acessórios",
    subcategories: [
      { code: "masculino", displayName: "Masculino" },
      { code: "feminino", displayName: "Feminino" },
      { code: "infantil", displayName: "Infantil" },
    ],
  },
  {
    code: "veiculos",
    displayName: "Veículos & Peças",
    subcategories: [
      { code: "carro", displayName: "Carro" },
      { code: "moto", displayName: "Moto" },
      { code: "pecas", displayName: "Peças" },
    ],
  },
  {
    code: "casa-jardim",
    displayName: "Casa & Jardim",
    subcategories: [
      { code: "ferramentas", displayName: "Ferramentas" },
      { code: "jardinagem", displayName: "Jardinagem" },
      { code: "eletrodomestico", displayName: "Eletrodoméstico" },
    ],
  },
  {
    code: "esportes",
    displayName: "Esportes & Lazer",
    subcategories: [
      { code: "bicicleta", displayName: "Bicicleta" },
      { code: "academia", displayName: "Academia" },
      { code: "camping", displayName: "Camping" },
    ],
  },
  {
    code: "infantil",
    displayName: "Infantil & Bebê",
    subcategories: [
      { code: "brinquedo", displayName: "Brinquedo" },
      { code: "carrinho", displayName: "Carrinho/Berço" },
      { code: "roupa", displayName: "Roupa" },
    ],
  },
  {
    code: "livros",
    displayName: "Livros & Revistas",
    subcategories: [
      { code: "ficcao", displayName: "Ficção" },
      { code: "didatico", displayName: "Didático" },
      { code: "outros", displayName: "Outros" },
    ],
  },
  {
    code: "servicos",
    displayName: "Serviços",
    subcategories: [
      { code: "reforma", displayName: "Reforma" },
      { code: "limpeza", displayName: "Limpeza" },
      { code: "aulas", displayName: "Aulas" },
    ],
  },
  {
    code: "outros",
    displayName: "Outros",
    subcategories: [{ code: "diversos", displayName: "Diversos" }],
  },
];

export const CATEGORY_CODES = CATEGORIES.map((c) => c.code);

export function findCategory(code: string): LocalCategory | undefined {
  return CATEGORIES.find((c) => c.code === code);
}

export function findSubcategories(categoryCode: string): LocalSubcategory[] {
  return findCategory(categoryCode)?.subcategories ?? [];
}

File: src/BairroNow.Api/Constants/Categories.cs
namespace BairroNow.Api.Constants;

// Phase 4 D-24, D-25: Hardcoded 10-category taxonomy with 2-3 subcategories per category.
// Admin can ON/OFF categories via toggles (D-26) -- Enabled flag is loaded/persisted by
// CategoriesController via an in-memory dictionary backed by IMemoryCache (no DB table in MVP).
public static class Categories
{
    public sealed record Subcategory(string Code, string DisplayName);
    public sealed record Category(string Code, string DisplayName, IReadOnlyList<Subcategory> Subcategories);

    public static readonly IReadOnlyList<Category> All = new List<Category>
    {
        new("eletronicos",  "Eletrônicos & Informática", new List<Subcategory>
        {
            new("celular",  "Celular"),
            new("notebook", "Notebook"),
            new("tv",       "TV"),
            new("outros",   "Outros"),
        }),
        new("moveis", "Móveis & Decoração", new List<Subcategory>
        {
            new("sala",    "Sala"),
            new("quarto",  "Quarto"),
            new("cozinha", "Cozinha"),
        }),
        new("roupas", "Roupas, Calçados & Acessórios", new List<Subcategory>
        {
            new("masculino",  "Masculino"),
            new("feminino",   "Feminino"),
            new("infantil",   "Infantil"),
        }),
        new("veiculos", "Veículos & Peças", new List<Subcategory>
        {
            new("carro",  "Carro"),
            new("moto",   "Moto"),
            new("pecas",  "Peças"),
        }),
        new("casa-jardim", "Casa & Jardim", new List<Subcategory>
        {
            new("ferramentas",   "Ferramentas"),
            new("jardinagem",    "Jardinagem"),
            new("eletrodomestico","Eletrodoméstico"),
        }),
        new("esportes", "Esportes & Lazer", new List<Subcategory>
        {
            new("bicicleta", "Bicicleta"),
            new("academia",  "Academia"),
            new("camping",   "Camping"),
        }),
        new("infantil", "Infantil & Bebê", new List<Subcategory>
        {
            new("brinquedo", "Brinquedo"),
            new("carrinho",  "Carrinho/Berço"),
            new("roupa",     "Roupa"),
        }),
        new("livros", "Livros & Revistas", new List<Subcategory>
        {
            new("ficcao",   "Ficção"),
            new("didatico", "Didático"),
            new("outros",   "Outros"),
        }),
        new("servicos", "Serviços", new List<Subcategory>
        {
            new("reforma",  "Reforma"),
            new("limpeza",  "Limpeza"),
            new("aulas",    "Aulas"),
        }),
        new("outros", "Outros", new List<Subcategory>
        {
            new("diversos", "Diversos"),
        }),
    };

    public static readonly string[] CATEGORY_CODES = All.Select(c => c.Code).ToArray();

    public static bool IsValidCategoryCode(string code) =>
        !string.IsNullOrWhiteSpace(code) && CATEGORY_CODES.Contains(code);

    public static bool IsValidSubcategoryCode(string categoryCode, string subCode)
    {
        var cat = All.FirstOrDefault(c => c.Code == categoryCode);
        return cat != null && cat.Subcategories.Any(s => s.Code == subCode);
    }
}
